docs/DeveloperGuide.md

```diff
@@ -1,13 +1,249 @@
+- author: Apache Commons
+- source: [https://commons.apache.org/proper/commons-lang/](https://commons.apache.org/proper/commons-lang/)
+
+**Alpha Vantage Stock Market API**
+- author: Alpha Vantage
+- source: [https://www.alphavantage.co/](https://www.alphavantage.co/)
+
+**Gson Java library**
+- author: Google
+- source: [https://github.com/google/gson](https://github.com/google/gson)
+
+**Financial Modeling Prep Stock API**
+- author: Financial Modeling Prep
+- source: https://site.financialmodelingprep.com/
+
+**round() method in Cashflow.java**
+  - author: mhadidg
+  - source: [https://stackoverflow.com/questions/2808535/round-a-double-to-2-decimal-places]()
+
+**capitalize() method in Cashflow.java**
+  - author: Nick Bolton
+  - source: [https://stackoverflow.com/questions/1892765/how-to-capitalize-the-first-character-of-each-word-in-a-string]()
+    
+**DG adapted from**
+
+* [Addressbook-level3](https://github.com/se-edu/addressbook-level3)
+
 {list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
 
 ## Design & implementation
 
 {Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
 
+### Storage Component
+
+API: `Storage.java`
+
+![](images/Storage.png)
+
+* The storage component loads data from the saved text files when the application starts, and saves the data to the
+  text files when the application exits.
+* The storage class uses the static methods in LoadData and SaveData to load and save data respectively.
+* The `load` method in LoadData reads the `data.txt` file and loads any existing Income, Expense and Budget into the application.
+* The `save` method in SaveData saves all Incomes, Expenses and existing Budget into the `data.txt` file.
+
+#### Design considerations:
+
+* There are 2 main ways to implement the storage, one is to save the data after every command, and the other is to save 
+the data one upon exiting the program with the `exit` command.
+* Saving the data once upon exit (Currently implemented):
+  * Advantage: Better efficiency and performance of the program.
+  * Disadvantage: If the program crashes or exits incorrectly, data will not be saved.
+* Saving the data after every command:
+  * Advantage: Changes are saved after every command.
+  * Disadvantage: Executing command might slow down the program when there is a large amount of data to be saved.
+
+### Visualization Feature 
+
+This feature is implemented with the help of [XChart](https://knowm.org/open-source/xchart/), a simple charting library for Java by Knowm.
+
+By typing in the vis command with the appropriate arguments (/s and /t), users will be able to visualize their income or expense 
+using visualization tools (Piechart, Bar Chart)
+
+Demo: 
+
+`vis /t expense /c pie`
+
+Output
+
+`Displaying piechart for expense`
+A message will be shown telling you that the chart is being displayed
```

Comment:
Really nice implementation. This feature is straightforward and meets the user requirements.